pkg/admission/util.go

```diff
@@ -29,42 +29,79 @@ import (
 	"github.com/apache/yunikorn-k8shim/pkg/log"
 )
 
-func updatePodLabel(pod *v1.Pod, namespace string, generateUniqueAppIds bool, defaultQueueName string) map[string]string {
-	existingLabels := pod.Labels
-	result := make(map[string]string)
-	for k, v := range existingLabels {
-		result[k] = v
-	}
+func getNewApplicationInfo(pod *v1.Pod, namespace string, generateUniqueAppIds bool, defaultQueueName string) (map[string]string, map[string]string) {
```

Comment:
In this function, I think it's useful to issue a warning to the user that labels will be deprecated in a later release, so they must adapt their app submission logic to this. It's very easy to forget about this and we must reduce the surprise factor.

What I'd do: return an extra bool from `getApplicationIDFromPod()`, `getDisableStateAwareFromPod()`, `utils.GetQueueNameFromPod()` to indicate whether a source was a label. If there's at least one of them is true, log something useful:
```
if appIdLabel || stateAwareFromLabel || queueNameFromLabel {
  log.Log(log.Admission).Warn("Pod contains label for Yunikorn configuration. This is deprecated and will be 
  ignored in a future release. Please migrate to annotation-based configuration.",
  zap.Bool("appId from label", appIdLabel),
  zap.Bool("stateAware from label", stateAwareFromLabel),
  zap.Bool("queue name from label", queueNameFromLabel))
}
```

Reply:
Add deprecated warning if applicationInfo is from labels (cf71eeff4cef7f432cc8778de11dfb14202efb91)

- Return extra bool 'isFromLabel' as suggested

Reply:
I don't think we can even deprecate the `applicationId` label -- it's far too entrenched in the existing ecosystem (and arguably it should be a label anyway to allow querying pods by applicationId). I think our best best is simply to use the annotation if it exists, and fallback to the label if it does not.

Reply:
It make sense to me, let's discuss the expecting AM behavior in below discussion.